backend/server.js

```diff
@@ -20,7 +31,36 @@ app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hi there Allan!')
+})
+
+// Sign up endpoint
+app.post('/signup', async (req, res) => {
+  try {
+    const { name, email, password } = req.body
+    const user = await new User({ name, email, password: bcrypt.hashSync(password) })
+    user.save()
+    res.status(201).json({ id: user._id, accessToken: user.accessToken })
+  } catch (err) {
+    res.status(400).json({ message: 'Could not create user!', errors: err.errors })
+  }
+})
```

Comment:
You should await the `save()` in order to solve the issue you had with double registration:
```
const user = new User({ name, email, password: bcrypt.hashSync(password) })
  const savedUser = await user.save()
  res.status(201).json({ id: savedUser._id, accessToken: savedUser.accessToken })
```